includes/class-newspack-popups-segmentation.php

```diff
@@ -343,6 +349,15 @@ public static function create_database_table() {
 		}
 	}
 
+	/**
+	 * Create default segments for sites that don't have any existing segments.
+	 */
+	public static function new_site_setup() {
+		if ( empty( self::get_segments() ) ) {
```

Comment:
Perhaps I'm overthinking this, but some users might just not want any segments. After the defaults are created, they'd delete those, and expect to have only the "Everyone" pseudo-segment in the UI. To make it possible, the check should be on an option that's saved when the segments are created for the first time. If they were created once, site should re-create the segments.

Reply:
I think you're right. 0aaae26 updates the function so that the default segments are created only if there are no segments AND the default segments haven't been created yet (i.e. the option doesn't exist).